framework/src/meshgenerators/AddMetaDataGenerator.C

```diff
@@ -0,0 +1,156 @@
+                                      _uint_scalar_metadata_values[i]);
+
+  inputChecker(_int_scalar_metadata_names, _int_scalar_metadata_values, "int_scalar");
+  for (unsigned int i = 0; i < _int_scalar_metadata_names.size(); i++)
+    declareMeshProperty<int>(_int_scalar_metadata_names[i], _int_scalar_metadata_values[i]);
+
+  inputChecker(_dof_id_type_scalar_metadata_names,
+               _dof_id_type_scalar_metadata_values,
+               "dof_id_type_scalar");
+  for (unsigned int i = 0; i < _dof_id_type_scalar_metadata_names.size(); i++)
+    declareMeshProperty<dof_id_type>(_dof_id_type_scalar_metadata_names[i],
+                                     _dof_id_type_scalar_metadata_values[i]);
+
+  inputChecker(_real_vector_metadata_names, _real_vector_metadata_values, "real_vector");
+  for (unsigned int i = 0; i < _real_vector_metadata_names.size(); i++)
+    declareMeshProperty<std::vector<Real>>(_real_vector_metadata_names[i],
+                                           _real_vector_metadata_values[i]);
+
+  inputChecker(_uint_vector_metadata_names, _uint_vector_metadata_values, "uint_vector");
+  for (unsigned int i = 0; i < _uint_vector_metadata_names.size(); i++)
+    declareMeshProperty<std::vector<unsigned int>>(_uint_vector_metadata_names[i],
+                                                   _uint_vector_metadata_values[i]);
+
+  inputChecker(_int_vector_metadata_names, _int_vector_metadata_values, "int_vector");
+  for (unsigned int i = 0; i < _int_vector_metadata_names.size(); i++)
+    declareMeshProperty<std::vector<int>>(_int_vector_metadata_names[i],
+                                          _int_vector_metadata_values[i]);
+
+  inputChecker(_dof_id_type_vector_metadata_names,
+               _dof_id_type_vector_metadata_values,
+               "dof_id_type_vector");
+  for (unsigned int i = 0; i < _dof_id_type_vector_metadata_names.size(); i++)
+    declareMeshProperty<std::vector<dof_id_type>>(_dof_id_type_vector_metadata_names[i],
+                                                  _dof_id_type_vector_metadata_values[i]);
+}
+
+std::unique_ptr<MeshBase>
+AddMetaDataGenerator::generate()
+{
+  std::unique_ptr<MeshBase> mesh = std::move(_input);
+
+  return dynamic_pointer_cast<MeshBase>(mesh);
+}
+
+template <class T>
+void
+AddMetaDataGenerator::inputChecker(const std::vector<std::string> data_names,
+                                   const std::vector<T> data_values,
+                                   const std::string param_name)
+{
+  std::vector<std::string> data_name_tmp(data_names);
+  if (std::unique(data_name_tmp.begin(), data_name_tmp.end()) != data_name_tmp.end())
+    paramError(param_name + "_metadata_names", "Elements of this parameter must be unique.");
+  if (data_names.size() != data_values.size())
+    paramError(param_name + "_metadata_values",
+               "Length of this parameter must be the same as that of " + param_name +
+                   "_metadata_names");
+}
```

Comment:
civet wont lack this missing newline

Reply:
Still struggling with transition from Atom to VSCode. 😞

Reply:
there's a vscode channel on slack btw where people share tips